### PR TITLE
handle all arc.js datetime types

### PR DESCRIPTION
**Pull request by dkent600**
Resolves: https://github.com/daostack/alchemy/issues/1984

* Adds `safeMoment`.
* a couple other tweaks to attempt to ensure we're safe from any changes to arc.js wrt datetime typing
  * where Competition computes its status with respect to its workflow
  * where the vote buttons compute whether the proposal was created before the account had rep in the DAO

### CODE REVIEW

**Comment by dkent600** [comment]
> So if we solve this issue from arc.js this will not be necessary?

I'm not sure at the moment, would need to research it a bit.  May depend on which data type is chosen.  May not hurt to keep the change, just to be robust.

This line in particular should be checked:

image

File: src/lib/util.ts [hunk 1]
```diff
@@ -301,7 +301,7 @@ export async function getNetworkId(web3Provider?: Web3Provider): Promise<string>
       return network.chainId.toString();
     } else {
       const promise = new Promise<string>((resolve, reject) => {
-        web3Provider.send("net_version", (error, response) => {
+        web3Provider.send("net_version", (error: any, response: any) => {
```
Comment (roienatan) [inline src/lib/util.ts:304]: Is it related?

Comment (dkent600) [reply src/lib/util.ts:304]: it is just the usual typescript lint warning improvements, an ongoing improvement...does not affect execution of the code.

Comment (roienatan) [reply src/lib/util.ts:304]: ok, yes I assume also the `boolean` in `proposalHelpers.ts`

**Pull request merged**